recirq/time_crystals/dtcexperiment_test.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2021 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import recirq.time_crystals as time_crystals
+import cirq
+import numpy as np
+import itertools
+
+
+def test_DTCExperiment():
+    np.random.seed(5)
+    qubit_locations = [
+        (3, 9),
+        (3, 8),
+        (3, 7),
+        (4, 7),
+        (4, 8),
+        (5, 8),
+        (5, 7),
+        (5, 6),
+        (6, 6),
+        (6, 5),
+        (7, 5),
+        (8, 5),
+        (8, 4),
+        (8, 3),
+        (7, 3),
+        (6, 3),
+    ]
+
+    qubits = [cirq.GridQubit(*idx) for idx in qubit_locations]
+    num_qubits = len(qubits)
+    g = 0.94
+    instances = 36
+    initial_state = np.random.choice(2, num_qubits)
+    local_fields = np.random.uniform(-1.0, 1.0, (instances, num_qubits))
+    thetas = np.zeros((instances, num_qubits - 1))
+    zetas = np.zeros((instances, num_qubits - 1))
+    chis = np.zeros((instances, num_qubits - 1))
+    gammas = -np.random.uniform(0.5 * np.pi, 1.5 * np.pi, (instances, num_qubits - 1))
+    phis = -2 * gammas
+    args = [
+        "qubits",
+        "g",
+        "initial_state",
+        "local_fields",
+        "thetas",
+        "zetas",
+        "chis",
+        "gammas",
+        "phis",
+    ]
+    default_resolvers = time_crystals.DTCExperiment().param_resolvers()
+    for arg in args:
+        kwargs = {}
+        for name in args:
+            kwargs[name] = None if name is arg else locals()[name]
```

Comment:
please don't be fancy like this with tests. Keep them "DAMP" (descriptive and meaningful phrases) rather than "DRY" (don't repeat yourself)

In addition to being clunky, this test is broken when I run it

Reply:
Tests should be working properly now, and the code is less fancy. 
What they _do_, however, hasn't changed. They still simply run every combination of default or supplied values for path coverage. This could be improved but I'm not sure how would be best.